From 5d6321460b02f193387ca06800e57d5bd3c74c2c Mon Sep 17 00:00:00 2001
From: Tendai Makumire <tendai.makumire@imd-tec.com>
Date: Fri, 28 Aug 2026 12:20:49 +0100
Subject: [PATCH] Seperate SBC8550 rev3 and rev5 into individual board DTs

Only the newer rev5 board has the CYPD which allows
us to configure DP. To allow us to support both
old and newer boards, we now have seperate
DTS files for the different revisions.
---
 Kbuild                                        |  3 +-
 ...ama-imdt-sde-display-hdk-rev3-overlay.dts} |  2 +-
 ...lama-imdt-sde-display-hdk-rev5-overlay.dts | 13 ++++++
 display/kalama-imdt-sde-display-hdk-rev5.dtsi | 27 +++++++++++
 display/kalama-imdt-sde-display-hdk.dtsi      | 45 +++++--------------
 5 files changed, 55 insertions(+), 35 deletions(-)
 rename display/{kalama-sde-display-hdk-overlay.dts => kalama-imdt-sde-display-hdk-rev3-overlay.dts} (89%)
 create mode 100755 display/kalama-imdt-sde-display-hdk-rev5-overlay.dts
 create mode 100755 display/kalama-imdt-sde-display-hdk-rev5.dtsi

diff --git a/Kbuild b/Kbuild
index 7f0e139..1b34f6d 100755
--- a/Kbuild
+++ b/Kbuild
@@ -5,7 +5,8 @@ dtbo-$(CONFIG_ARCH_KALAMA) += display/kalama-sde.dtbo \
 		display/kalama-sde-display-cdp-wsa883x-overlay.dtbo \
 		display/kalama-sde-display-mtp-overlay.dtbo \
 		display/kalama-sde-display-qrd-overlay.dtbo \
-		display/kalama-sde-display-hdk-overlay.dtbo \
+		display/kalama-imdt-sde-display-hdk-rev5-overlay.dtbo \
+		display/kalama-imdt-sde-display-hdk-rev3-overlay.dtbo \
 		display/kalama-sde-display-rb5-gen2-overlay.dtbo \
 		display/kalama-sde-display-aim300-overlay.dtbo \
 		display/kalama-sde-display-hhg-overlay.dtbo \
diff --git a/display/kalama-sde-display-hdk-overlay.dts b/display/kalama-imdt-sde-display-hdk-rev3-overlay.dts
similarity index 89%
rename from display/kalama-sde-display-hdk-overlay.dts
rename to display/kalama-imdt-sde-display-hdk-rev3-overlay.dts
index 7074556..76bd665 100755
--- a/display/kalama-sde-display-hdk-overlay.dts
+++ b/display/kalama-imdt-sde-display-hdk-rev3-overlay.dts
@@ -9,5 +9,5 @@
 			"qcom,kalamap";
 	qcom,msm-id = <536 0x10000>, <536 0x20000>, <603 0x20000>,
 			<604 0x20000>;
-	qcom,board-id = <0x1001f 0>;
+	qcom,board-id = <0x0001001f 0>;
 };
diff --git a/display/kalama-imdt-sde-display-hdk-rev5-overlay.dts b/display/kalama-imdt-sde-display-hdk-rev5-overlay.dts
new file mode 100755
index 0000000..c7e0b12
--- /dev/null
+++ b/display/kalama-imdt-sde-display-hdk-rev5-overlay.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+#include "kalama-imdt-sde-display-hdk-rev5.dtsi"
+
+/ {
+	model = "Qualcomm Technologies, Inc. Kalama HDK";
+	compatible = "qcom,kalama-hdk", "qcom,kalama", "qcom,hdk",
+			"qcom,kalamap";
+	qcom,msm-id = <536 0x10000>, <536 0x20000>, <603 0x20000>,
+			<604 0x20000>;
+	qcom,board-id = <0x0101001f 0>;
+};
diff --git a/display/kalama-imdt-sde-display-hdk-rev5.dtsi b/display/kalama-imdt-sde-display-hdk-rev5.dtsi
new file mode 100755
index 0000000..92ab24b
--- /dev/null
+++ b/display/kalama-imdt-sde-display-hdk-rev5.dtsi
@@ -0,0 +1,27 @@
+#include "kalama-imdt-sde-display-hdk.dtsi"
+
+&sde_dp {
+	status = "okay";
+
+	mode-switch;
+	svid = /bits/ 16 <0xFF01>;
+
+	port {
+		dp_altmode_ep: endpoint {
+			remote-endpoint = <&typec_connector_port1_ep>;
+		};
+	};
+};
+
+&typec_connector_ports {
+	port@1 {
+		reg = <1>;
+		typec_connector_port1_ep: endpoint {
+			remote-endpoint = <&dp_altmode_ep>;
+		};
+	};
+};
+
+&mdss_mdp {
+	connectors = <&sde_dsi &smmu_sde_unsec &sde_rscc &sde_wb1 &sde_dp>;
+};
diff --git a/display/kalama-imdt-sde-display-hdk.dtsi b/display/kalama-imdt-sde-display-hdk.dtsi
index d748738..8a66593 100755
--- a/display/kalama-imdt-sde-display-hdk.dtsi
+++ b/display/kalama-imdt-sde-display-hdk.dtsi
@@ -37,38 +37,17 @@
 	status = "disabled";
 };
 
-&sde_dp {
-	status = "okay";
-
-	mode-switch;
-	svid = /bits/ 16 <0xFF01>;
-
-	port {
-		dp_altmode_ep: endpoint {
-			remote-endpoint = <&typec_connector_port1_ep>;
-		};
-	};
-};
-
-&typec_connector_ports {
-	port@1 {
-		reg = <1>;
-		typec_connector_port1_ep: endpoint {
-			remote-endpoint = <&dp_altmode_ep>;
-		};
-	};
-};
-
-/*
- * We must remove the connectors/components we are
- * are not using from the list. In our case,
- * we are currently not using DSI1.
- * So sde_dsi1 (DSI1) has been removed from our list of connectors.
- *
- * The display kernel drivers works similarly to the camera
- * kernel drivers where they use the Component Helper
- * subsystem to wait for multiple subdevices to be probed.
- */
+/* 	IMDT NOTE: 	
+	We must remove the connectors/components we are
+	are not using from the list. In our case,
+	we are currently not using DSI1 and the Display Port.
+	So &sde_dsi1 (DSI1) and &sde_dp (Display Port) have not 
+	be included in our list of connectors.
+
+	The display kernel drivers works similarly to the camera
+	kernel drivers where they use the Component Helper
+	subsystem to wait for multiple subdevices to be probed.
+*/
 &mdss_mdp {
-	connectors = <&sde_dsi &smmu_sde_unsec &sde_rscc &sde_wb1 &sde_dp>;
+	connectors = <&sde_dsi &smmu_sde_unsec &sde_rscc &sde_wb1>;
 };
-- 
2.41.0

